newIDE/app/src/MainFrame/Onboarding/OnboardingDialog.js

```diff
@@ -0,0 +1,122 @@
+// @flow
+import { Trans } from '@lingui/macro';
+
+import * as React from 'react';
+import userflow from 'userflow.js';
+import Dialog from '../../UI/Dialog';
+import FlatButton from '../../UI/FlatButton';
+import optionalRequire from '../../Utils/OptionalRequire';
+import { getProgramOpeningCount } from '../../Utils/Analytics/LocalStats';
+import { isMobile } from '../../Utils/Platform';
+import Window from '../../Utils/Window';
+import { loadPreferencesFromLocalStorage } from '../Preferences/PreferencesProvider';
+import { getUserUUID } from '../../Utils/Analytics/UserUUID';
+import RaisedButton from '../../UI/RaisedButton';
+import { Column, Line } from '../../UI/Grid';
+import { MarkdownText } from '../../UI/MarkdownText';
+import { showErrorBox } from '../../UI/Messages/MessageBox';
+
+const electron = optionalRequire('electron');
+const isDev = Window.isDev();
+export let isUserflowRunning = false;
+
+const onboardingText = `
+In 5 minutes, you will have:
+  - Created a game
+  - Learned the fundamentals of GDevelop
+
+(🇬🇧 The tour is only available in English)  
+We highly recommend it!
+`;
+
+const OnboardingDialog = () => {
+  const [open, setOpen] = React.useState(false);
+  const [loading, setLoading] = React.useState(false);
+
+  React.useEffect(() => {
+    userflow.on(
```

Comment:
Let's hope this work even if init was not called? Might be worth checking that or moving that inside startUserflow?

Also, this global callback on userflow will be added every time this dialog is shown.
Sure for now it's only once, but as soon as we add a button "Follow the tour" in the homepage that will show this dialog, we start to add more and more `on('flowvisibilitychange'` and that seems risky.

I would do two things (maybe confirm with Alex):
- only update the loading and open state in `startUserflow`.
- so you keep only `isUserflowRunning = isRunning` inside the callback. If you "leak" the callback by creating too much of them, it's not a huge problem but still dirty and a receipt for future problems if someone adds additional logic. So you can add a global boolean "isUserflowEventListenerSetUp" to false by default, and run this `userflow.on` only when the `isUserflowEventListenerSetUp`is false, immediately setting it to `true` after. 

If you do this after the `init`, then we are:
- feeling safer (we don't call any method before userflow.init... I'm always super suspicious about the capacity of 3rd party libraries to be coded well enough to be robust against this).
- don't create a leak of listeners if we show the dialog multiple times.

Reply:
Agree we can be more robust.
I tested it locally and the `userflow.on()` was correctly registered, even if the init happened afterwards (the modal closed + the flow started correctly)

But I'll improve that